### PR TITLE
Dockerfile formatting changes and optimisations

### PR DESCRIPTION
**Pull request by thaJeztah**
follow-up changes to https://github.com/moby/moby/pull/39340 (for discussion, see individual commits for what changed)

### CODE REVIEW

**Comment by thaJeztah** [comment]
@AkihiroSuda @cpuguy83 PTAL

File: Dockerfile [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ARG CROSS="false"
 ARG GO_VERSION=1.13.1
 ARG DEBIAN_FRONTEND=noninteractive
+ARG VPNKIT_DIGEST=e508a17cfacc8fd39261d5b4e397df2b953690da577e2c987a47630cd0c42f8e
```
Comment (cpuguy83) [inline Dockerfile]: This doesn't seem cross platform.

Comment (thaJeztah) [reply Dockerfile]: Oh! Now I recall why we did it the way it was 😞

Comment (thaJeztah) [reply Dockerfile]: Actually, I think that should work, as we currently also pull it unconditionally?

Comment (cpuguy83) [reply Dockerfile]: But the image is for a specific architecture (amd64, I'm assuming).

Comment (cpuguy83) [reply Dockerfile]: I guess it's not really a change from what we already have.

Comment (thaJeztah) [reply Dockerfile]: Yes, looks like we worked around it at the time by naming the binary `something.x86`. I guess with BuildKit we could make an architecture-specific dummy stage, but not sure if that would make things a lot cleaner

**Comment by thaJeztah** [comment]
@cpuguy83 @akihirosuda PTAL; planning to make changes after this one to bring the versions into the dockerfile itself (and some other changes)

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Review by kolyshkin — APPROVED** [review]
LGTM

**Pull request merged**